File: VB/WebSite/Controllers/HomeController.vb
Imports Microsoft.VisualBasic
Imports System
Imports System.Collections.Generic
Imports System.Linq
Imports System.Web
Imports System.Web.Mvc
Imports DevExpress.Web.Mvc
Imports DevExpress.XtraPivotGrid
Imports DevExpress.XtraPrinting
Imports DevExpress.XtraPrintingLinks
Imports System.IO
Imports System.Reflection
Imports System.Drawing

Namespace DevExpressMvcApplication17.Controllers
	Public Class HomeController
		Inherits Controller
		Public Function Index() As ActionResult
			ViewBag.Message = "Welcome to DevExpress Extensions for ASP.NET MVC!"
			Return View()
		End Function
		Public Function PivotGridPartial() As ActionResult
			Return PartialView("PivotGridPartial")
		End Function

		<HttpPost> _
		Public Function PostPivotGridState() As ActionResult
			ViewBag.Message = "The PivotGrid state has been restored successfully"
			Return View("Index")
		End Function
	End Class
End Namespace

File: CS/WebSite/Controllers/HomeController.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Web;
using System.Web.Mvc;
using DevExpress.Web.Mvc;
using DevExpress.XtraPivotGrid;
using DevExpress.XtraPrinting;
using DevExpress.XtraPrintingLinks;
using System.IO;
using System.Reflection;
using System.Drawing;

namespace DevExpressMvcApplication17.Controllers {
    public class HomeController : Controller {
        public ActionResult Index() {
            ViewBag.Message = "Welcome to DevExpress Extensions for ASP.NET MVC!";
            return View();
        }
        public ActionResult PivotGridPartial() {
            return PartialView("PivotGridPartial");
        }

        [HttpPost]
        public ActionResult PostPivotGridState() {
            ViewBag.Message = "The PivotGrid state has been restored successfully";
            return View("Index");
        }
    }
}